src/main/java/com/minecraftabnormals/abnormals_core/core/mixin/EndBiomeProviderMixin.java

```diff
@@ -54,23 +45,21 @@ private void init(Registry<Biome> lookupRegistry, long seed, CallbackInfo info)
 		this.noiseBiomeLayer = ACLayerUtil.createEndBiomeLayer(lookupRegistry, (seedModifier) -> new LazyAreaLayerContext(25, seed, seedModifier));
 	}
 
-	@Inject(at = @At("HEAD"), method = "getNoiseBiome(III)Lnet/minecraft/world/biome/Biome;", cancellable = true)
+	@Inject(at = @At("RETURN"), method = "getNoiseBiome(III)Lnet/minecraft/world/biome/Biome;", cancellable = true)
 	private void addEndBiomes(int x, int y, int z, CallbackInfoReturnable<Biome> info) {
-		int i = x >> 2;
-		int j = z >> 2;
-		if ((long) i * (long) i + (long) j * (long) j <= 4096L) {
-			info.setReturnValue(this.theEndBiome);
-		} else {
-			float noise = EndBiomeProvider.getRandomNoise(this.generator, i * 2 + 1, j * 2 + 1);
-			Biome biome = getNoiseBiome(x, z);
-			boolean isChorus = biome == this.endMidlandsBiome;
-			if (noise > 40.0F) {
-				info.setReturnValue(isChorus ? this.endHighlandsBiome : biome);
-			} else if (noise >= 0.0F) {
-				info.setReturnValue(isChorus ? this.endMidlandsBiome : biome);
-			} else {
-				info.setReturnValue(noise < -20.0F ? this.smallEndIslandsBiome : isChorus ? this.endBarrensBiome : biome);
-			}
+		Biome oldBiome = (Biome)info.getReturnValue();
+		Biome newBiome = getNoiseBiome(x, z);
```

Comment:
Not present in the old code, but would be a nice style update.
```suggestion
		Biome newBiome = this.getNoiseBiome(x, z);
```